core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -143,6 +143,28 @@ class RemoteIndexCacheTest {
     verifyNoInteractions(rsm)
   }
 
+  @Test
+  def testFetchIndexForMissingTransactionIndex(): Unit = {
+    when(rsm.fetchIndex(any(classOf[RemoteLogSegmentMetadata]), any(classOf[IndexType])))
```

Comment:
I may be missing something here. How is this test validating the actual implementation?

Reply:
In this test, I am trying to verify that if the actual implementation throws the `RemoteResourceNotFoundException`, it is handled correctly by the `createCacheEntry` method.